### ISSUE TITLE
Another Kind Issue?

### ISSUE THREAD

**Issue by magnus-madsen** [flix/flix#1934]
This:

```
enum Jazz[a] {}

instance Order[Jazz] {
    pub def compare(xs: Jazz[a], ys: Jazz[a]): Comparison = ???
}
```

gives this:

```
-- Instance Error -------------------------------------------------- foo3.flix


Mismatched signature 'compare' required by class 'Order'.

4 |     pub def compare(xs: Jazz[a], ys: Jazz[a]): Comparison = ???
                ^^^^^^^
                mismatched signature.


Expected scheme: Jazz -> Jazz -> Comparison with [Jazz : Order]
Actual scheme:   ∀(a). Jazz[a] -> Jazz[a] -> Comparison with [Jazz : Order]

Tip: Modify the definition to match the signature.

Compilation failed with 1 error(s).
```

but I would have expected an error near `Order[Jazz]` (Since the kind of Order is Type and not Type -> Type).

**Comment by magnus-madsen** [comment]
At least it does give us an error and not a crash :)

**Comment by mlutze** [comment]
Fixed by #2037 

```
-- Kind Error -------------------------------------------------- ???

>> Expected kind 'Type' here, but kind 'Type -> Type' is used.

4 | instance Order[Jazz] {
                   ^^^^
                   unexpected kind.


Expected kind: Type
Actual kind:   Type -> Type
```

**Comment by magnus-madsen** [comment]
Fixed in master!
